realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -788,57 +788,101 @@ private Scanner getFullStringScanner(InputStream in) {
     /**
      * Insert a list of an unmanaged RealmObjects. This is generally faster than {@link #copyToRealm(Iterable)} since it
      * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * After being inserted any changes to the original objects will not be persisted.
      * <p>
-     * Please note, this doesn't check if the object is already managed or not, so inserting a managed object will duplicate it unlike {@link #copyToRealm(Iterable)}
-     * Copying an object will copy all field values. Any unset field in the object and child objects will be
-     * set to their default value if not provided.
+     * Please note:
+     * <ul>
+     * <li>We don't check if the provided objects are already managed or not, so inserting a managed object will duplicate it</li>
+     * <li>We don't create (nor return) a managed {@link RealmObject} for each element</li>
+     * <li>Copying an object will copy all field values. Any unset field in the object and child objects will be set to their default value if not provided</li>
+     * </ul>
+     * <p>
+     * If you want the previous checks and the returned managed {@link RealmObject} use {@link #copyToRealm(Iterable)}, otherwise if
+     * you have a large number of object this method is generally faster.
      *
      * @param objects RealmObjects to insert.
      */
     public void insert(Collection<? extends RealmModel> objects) {
+        checkIfValid();
```

Comment:
I guess the argument for having insert() for one object is that you use this in a loop perhaps? If you do that how much faster is this method now? Maybe it's sufficiently valuable that it creates less objects to GC?
Is the one object variants at all relevant to have still? Or wouldn't the collection variant be enough?